theme/theme.inc

```diff
@@ -112,3 +112,16 @@ function theme_islandora_solr_facet_wrapper($variables) {
   $output .= '</div>';
   return $output;
 }
+
+/**
+ * Implements hook_process_islandora_solr_search_navigation_block().
+ */
+function template_process_islandora_solr_search_navigation_block(&$variables) {
+  if ($variables['prev_link']) {
+    $variables['prev_link'] = l($variables['prev_text'], $variables['prev_link']);
+  }
+  $variables['return_link'] = l($variables['return_text'], $variables['return_link']);
+  if ($variables['next_link']) {
+    $variables['next_link'] = l($variables['next_text'], $variables['next_link']);
```

Comment:
Not completely comfortable here... `l()` calls `url()` inside of it, so we pass something which has already gone through `url()` through `url()` again... seems we have gotten into trouble due to this before, with stuff getting double-encoded and the like... Thinking building the markup by hand might be the way to go here?